src/boundary/boundary_p.cc

```diff
@@ -326,28 +332,71 @@ boundary_p( particle_bc_t       * RESTRICT pbc_list,
       
       n = sp->np + max_inj;
       if( n>sp->max_np ) {
-        n = n + (n>>2) + (n>>4); // Increase by 31.25% (~<"silver
+        n += 0.3125*n; // Increase by 31.25% (~<"silver
         /**/                     // ratio") to minimize resizes (max
         /**/                     // rate that avoids excessive heap
         /**/                     // fragmentation)
+        //float resize_ratio = (float)n/sp->max_np;
         WARNING(( "Resizing local %s particle storage from %i to %i",
                   sp->name, sp->max_np, n ));
         MALLOC_ALIGNED( new_p, n, 128 );
         COPY( new_p, sp->p, sp->np );
         FREE_ALIGNED( sp->p );
         sp->p = new_p, sp->max_np = n;
-      }
-      
-      n = sp->nm + max_inj;
-      if( n>sp->max_nm ) {
-        n = n + (n>>2) + (n>>4); // See note above
+
+        /*nm = sp->max_nm * resize_ratio;
         WARNING(( "Resizing local %s mover storage from %i to %i",
-                  sp->name, sp->max_nm, n ));
-        MALLOC_ALIGNED( new_pm, n, 128 );
+                  sp->name, sp->max_nm, nm ));
+        MALLOC_ALIGNED( new_pm, nm, 128 );
         COPY( new_pm, sp->pm, sp->nm );
         FREE_ALIGNED( sp->pm );
         sp->pm = new_pm;
-        sp->max_nm = n;
+        sp->max_nm = nm;*/
+      }
+// 32768 particles is 1 MiB of memory.
```

Comment:
Is this just arbitrary?

Reply:
I put in the minimum when I was trying to solve problems with resizing the movers (not in this pull request), but resizing the particle arrays to 0 (and not the mover arrays) does not seem to break anything.  I left in the minimum to avoid potential excessive resizing when there are few particles on a node.  Since the resizing is based on the relative needed/excess memory, it takes only a few particles crossing boundaries to trigger resizing on a nearly empty node.

I wanted a number high enough to avoid that and low enough to not waste a significant amount of RAM.  Beyond that, this number is arbitrary.  You can likely get rid of the minimum entirely with little performance impact (but even more warnings).

Reply:
OK cool 👍 

I think the minimum is a good idea, I just wanted to improve my understanding.

I wonder if a) the default should be something more like a page size, or a multiple of some number of particles, and b) if we should directly expose that to the user to be configurable? 

I should also add a nice user friendly way to turn the feature on/off

Reply:
OK sounds sensible. 

I've added code locally to expose the option nicely to the user. 

Can you send me a (very) small deck that should drop particles if the resizing isn't on? I'm hoping to add a automatic test